src/BigNumber.php

```diff
@@ -103,46 +98,63 @@ public static function of(BigNumber|int|float|string $value) : BigNumber
                 new BigInteger($denominator),
                 false
             );
-        }
+        } else {
+            // Integer or decimal number
+            if (\preg_match(self::PARSE_REGEXP_NUMERICAL, $value, $matches, PREG_UNMATCHED_AS_NULL) !== 1) {
+                self::throwException($value);
+            }
 
-        $point      = $getMatch('point');
-        $integral   = $getMatch('integral');
-        $fractional = $getMatch('fractional');
-        $exponent   = $getMatch('exponent');
+            $sign = $matches['sign'];
+            $point = $matches['point'];
+            $integral = $matches['integral'];
+            $fractional = $matches['fractional'];
+            $exponent = $matches['exponent'];
 
-        if ($integral === null && $fractional === null) {
-            $throw();
-        }
+            if ($integral === null && $fractional === null) {
+                self::throwException($value);
+            }
 
-        if ($integral === null) {
-            $integral = '0';
-        }
+            if ($integral === null) {
+                $integral = '0';
+            }
 
-        if ($point !== null || $exponent !== null) {
-            $fractional = ($fractional ?? '');
-            $exponent = ($exponent !== null) ? (int) $exponent : 0;
+            if ($point !== null || $exponent !== null) {
+                $fractional = ($fractional ?? '');
+                $exponent = ($exponent !== null) ? (int)$exponent : 0;
 
-            if ($exponent === PHP_INT_MIN || $exponent === PHP_INT_MAX) {
-                throw new NumberFormatException('Exponent too large.');
-            }
+                if ($exponent === PHP_INT_MIN || $exponent === PHP_INT_MAX) {
+                    throw new NumberFormatException('Exponent too large.');
+                }
 
-            $unscaledValue = self::cleanUp(($sign ?? ''). $integral . $fractional);
+                $unscaledValue = self::leanCleanUp(($sign ?? ''), $integral . $fractional);
```

Comment:
```suggestion
                $unscaledValue = self::cleanUp($sign, $integral . $fractional);
```

After `leanCleanUp` has been renamed to `cleanUp`.